url.py: Replace urlgrabber with a simple syscall to wget

sknsean opened the pull request:
The underlying problem is that pycurl in ubuntu and debian based
systems are linked with gnutls (instead of openssl)
This results in rror: gnutls_handshake() failed

Signed-off-by: Sean Nyekjaer sean.nyekjaer@prevas.dk

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
+    if not os.path.exists(os.path.dirname(filename)):
+        os.makedirs(os.path.dirname(filename))
+    print url + " " + filename
+    cmd = "/usr/bin/env wget -t 2 -T 30 --passive-ftp --no-check-certificate --progress=dot -v " + url + " -O " + filename
```
Use the timeout (`timeout`) and retries (`retry`) values from the arguments to `grab()`
Quote `url` and `filename`, or better yet: make the cmd variable a list of arguments: `cmd = ['/usr/bin/env', 'wget', ...]`

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
+
+    if not os.path.exists(os.path.dirname(filename)):
+        os.makedirs(os.path.dirname(filename))
+    print url + " " + filename
```
Put some context in this print and/or use oe-lite logging, e.g. the `info` imported in line 6.

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
+    print url + " " + filename
+    cmd = "/usr/bin/env wget -t 2 -T 30 --passive-ftp --no-check-certificate --progress=dot -v " + url + " -O " + filename
+
+    pipe = subprocess.Popen(cmd, shell=True,stderr=subprocess.PIPE, stdout=subprocess.PIPE)
```
`subprocess.Popen` doesn't return a `pipe` object, but a `Popen` object. It is usually named `p`, or `process`.
I don't think `shell=True` is needed (and it lacks a space after the comma).

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
+    stdout, stderr = pipe.communicate()
+
+    if pipe.returncode != 0:
+        print "Error", cmd, pipe.returncode, " ", stdout, stderr
```
Again, use oe-lite logging: `err()`

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
         return False
+
+    if not os.path.exists(filename):
+        print("The fetch command returned success for url %s but %s doesn't exist?!" % (uri, ud.localpath), uri)
```
oe-lite logging

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
+
+    if os.path.getsize(filename) == 0:
+        os.remove(filename)
+        print("The fetch of %s resulted in a zero size file?! Deleting and failing since this isn't right." % (uri), uri)
```
oe-lite logging

Commit:
url.py: Replace urlgrabber with a simple syscall to wget

The underlying problem is that pycurl in ubuntu and debian based
systems are linked with gnutls (instead of openssl)
This results in rror: gnutls_handshake() failed

Signed-off-by: Sean Nyekjaer <sean.nyekjaer@prevas.dk>

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
+    stdout, stderr = p.communicate()
+
+    if p.returncode != 0:
+        oebakery.err("Error %s %d %s %s" % ( cmd, p.returncode, stdout, stderr))
```
did you test this error path? I am pretty certain it fails due to the missing import of `oebakery`. Use plain `err()` instead.

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
         return False
+
+    if not os.path.exists(filename):
+        oebakery.err("The fetch command returned success for url %s but %s doesn't exist?!" % (uri, ud.localpath), uri)
```
same here

mnhu commented on lib/oelite/fetch/url.py:
```diff
@@ -116,34 +117,34 @@ def get_disable_ftp_epsv(self, d):
+
+    if os.path.getsize(filename) == 0:
+        os.remove(filename)
+        oebakery.err("The fetch of %s resulted in a zero size file?! Deleting and failing since this isn't right." % (uri), uri)
```
and here
